src/components/Register/Register.tsx

```diff
@@ -1,286 +1,94 @@
-                                            </button>
-                                        </div>
-                                    </span>
-                                );
-                            })()}
+                <div className="header">
+                    <span>Register</span>
+                </div>
+                <div className="register-content">
+                    <div className="first-page flex-column-box">
+                        <div className="inputs flex-column-box">
+                            <TextInput
+                                placeholder="E-Mail"
+                                onChange={(value) => registerStore.setEmail(value.target.value)}
+                                className="text-input"
+                                height="2.69rem"
+                                width="28rem"
+                            />
+                            <TextInput
+                                placeholder="Passwort"
+                                onChange={(value) => registerStore.setPassword(value.target.value)}
+                                className="text-input"
+                                height="2.69rem"
+                                width="28rem"
+                            />
+                            <TextInput
+                                placeholder="Passwort bestätigen"
+                                onChange={(value) => registerStore.setRetypedPassword(value.target.value)}
+                                className="text-input"
+                                height="2.69rem"
+                                width="28rem"
+                            />
+                            <div className="account-type flex-row-box">
+                                <Button
+                                    name="Lehrer"
+                                    onClick={() => registerStore.setAccountType('teacher')}
+                                    width="13.5rem"
+                                    height="2.69rem"
+                                    fontSize="22px"
+                                    styleType="unset"
+                                />
+                                <Button
+                                    name="Schüler"
+                                    onClick={() => registerStore.setAccountType('pupil')}
+                                    width="13.5rem"
+                                    height="2.69rem"
+                                    fontSize="22px"
+                                    styleType="unset"
+                                />
+                            </div>
                         </div>
-                    </form>
+                        <div className="actions flex-row-box">
+                            <input
+                                type="checkbox"
+                                onClick={() => registerStore.setIsAgbAccepted(registerStore.isAgbAccepted)}
```

Comment:
I think here is a logical mistake did you mean ```onClick={() => registerStore.setIsAgbAccepted(!registerStore.isAgbAccepted)}```? Because you want to toggle the value?
and rename Agb to Gtc (like below)